src/suggest/date-suggest.ts

```diff
@@ -104,19 +125,76 @@ export default class DateSuggest extends CodeMirrorSuggest<IDateCompletion> {
   ): void {
     const includeAlias = event.shiftKey;
 
-    const head = this.getStartPos();
-    const anchor = this.cmEditor.getCursor();
+    const { workspace } = this.app;
+    const activeView = workspace.getActiveViewOfType(MarkdownView);
+
+    if (!activeView) {
+      return;
+    }
+    const editor = activeView.editor;
+    const head = this.startPos;
+    const anchor = editor.getCursor();
+    let dateStr = "";
+    let makeIntoLink = this.plugin.settings.autosuggestToggleLink;
+
+    if (suggestion.label.startsWith("time:")) {
+      const timePart = suggestion.label.substring(5);
+      dateStr = this.plugin.parseTime(timePart).formattedString;
+      makeIntoLink = false;
+    } else {
+      dateStr = this.plugin.parseDate(suggestion.label).formattedString;
+    }
 
-    let dateStr = this.plugin.parseDate(suggestion.label).formattedString;
-    if (this.plugin.settings.autosuggestToggleLink) {
+    if (makeIntoLink) {
       if (includeAlias) {
         dateStr = `[[${dateStr}|${suggestion.label}]]`;
       } else {
         dateStr = `[[${dateStr}]]`;
       }
     }
 
-    this.cmEditor.replaceRange(dateStr, head, anchor);
+    editor.replaceRange(dateStr, head, anchor);
     this.close();
   }
+
+  onTrigger(
+    cursor: EditorPosition,
+    editor: Editor,
+    file: TFile
+  ): EditorSuggestTriggerInfo {
+    const lineContents = editor.getLine(cursor.line);
```

Comment:
I'm pretty sure this function be more similar to checkForInputPhrase worked previously. The only real change is instead of returning `true`, if we detect the input phrase, we return need to return `getTriggerInfo()`.